src/combinatorics/quiver/path.lean

```diff
@@ -30,7 +30,19 @@ path.nil.cons e
 
 namespace path
 
-variables {V : Type u} [quiver V] {a b c : V}
+variables {V : Type u} [quiver V] {a b c d : V}
+
+lemma nil_ne_cons (p : path a b) (e : b ⟶ a) : path.nil ≠ p.cons e.
+lemma cons_ne_nil (p : path a b) (e : b ⟶ a) : p.cons e ≠ path.nil.
+
+lemma obj_eq_of_cons_eq_cons {p : path a b} {p' : path a c}
+  {e : b ⟶ d} {e' : c ⟶ d} (h : p.cons e = p'.cons e') : b = c := by injection h
+
+lemma heq_of_cons_eq_cons {p : path a b} {p' : path a c}
+  {e : b ⟶ d} {e' : c ⟶ d} (h : p.cons e = p'.cons e') : p == p' := by injection h
```

Comment:
You have `obj_eq_of_cons_eq_cons`, so could you also give a `path.cast` version? Please also give a `hom.cast` version of the following lemma.

The `hom.cast` and `path.cast` versions are more powerful because they make equalities on indices immediately available. The `heq` terms do not imply equalities on indices.